TombEngine/Objects/TR5/Entity/tr5_imp.cpp

```diff
@@ -96,206 +135,252 @@ namespace TEN::Entities::Creatures::TR5
-					creature->Mood = MoodType::Escape;
+			angle = CreatureTurn(item, creature->MaxTurn);
 
-				CreatureMood(item, &AI, true);
+			if (Wibble & 0x10)
+				item->SetMeshSwapFlags(0x400);
+			else
+				item->SetMeshSwapFlags(NO_JOINT_BITS);
 
-				angle1 = CreatureTurn(item, creature->MaxTurn);
-				joint0 = AI.xAngle / 2;
-				joint1 = AI.angle / 2;
-				joint2 = AI.xAngle / 2;
-				joint3 = AI.angle / 2;
+			if (AI.ahead)
+			{
+				torso_x = AI.xAngle / 2;
+				torso_y = AI.angle / 2;
+				head_x = AI.xAngle / 2;
+				head_y = AI.angle / 2;
+			}
 
-				if (Wibble & 0x10)
-					item->SetMeshSwapFlags(1024);
-				else
-					item->SetMeshSwapFlags(NO_JOINT_BITS);
+			bool isScared = IsTorchLitNearby(item, creature);
+			switch (item->Animation.ActiveState)
+			{
+			case IMP_STATE_WALK:
+				creature->MaxTurn = IMP_RUN_TURN_RATE_MAX;
 
-				switch (item->Animation.ActiveState)
+				if (isScared)
 				{
-				case IMP_STATE_WALK:
-					creature->MaxTurn = ANGLE(7.0f);
-					if (AI.distance <= pow(SECTOR(2), 2))
-					{
-						if (AI.distance < pow(SECTOR(2), 2))
-							item->Animation.TargetState = IMP_STATE_IDLE;
-					}
-					else
-						item->Animation.TargetState = IMP_STATE_RUN;
-
+					item->Animation.TargetState = IMP_STATE_SCARED;
 					break;
+				}
 
-				case IMP_STATE_IDLE:
-					creature->MaxTurn = -1;
-					creature->Flags = 0;
-
-					if (AI.bite && AI.distance < pow(170, 2) && item->TriggerFlags < 10)
-					{
-						if (GetRandomControl() & 1)
-							item->Animation.TargetState = IMP_STATE_ATTACK_1;
-						else
-							item->Animation.TargetState = IMP_STATE_ATTACK_2;
-					}
-					else if (item->AIBits & FOLLOW)
-						item->Animation.TargetState = IMP_STATE_WALK;
-					else
-					{
-						if (item->TriggerFlags == 3)
-							item->Animation.TargetState = IMP_STATE_THROW_STONES;
-						else if (AI.distance <= pow(SECTOR(2), 2))
-						{
-							if (AI.distance > pow(SECTOR(0.5f), 2) || item->TriggerFlags < 10)
-								item->Animation.TargetState = IMP_STATE_WALK;
-						}
-						else
-							item->Animation.TargetState = IMP_STATE_RUN;
-					}
-
-					break;
+				if (AI.distance > IMP_WALK_RANGE)
+					item->Animation.TargetState = IMP_STATE_RUN;
+				else if (AI.distance < IMP_LOW_ATTACK_RANGE)
+					item->Animation.TargetState = IMP_STATE_IDLE;
 
-				case IMP_STATE_RUN:
-					creature->MaxTurn = ANGLE(7.0f);
+				break;
 
-					if (AI.distance >= pow(SECTOR(0.5f), 2))
-					{
-						if (AI.distance < pow(SECTOR(2), 2))
-							item->Animation.TargetState = IMP_STATE_WALK;
-					}
-					else
-						item->Animation.TargetState = IMP_STATE_IDLE;
+			case IMP_STATE_IDLE:
+				creature->MaxTurn = 0;
+				creature->Flags = 0;
 
+				if (isScared)
+				{
+					item->Animation.TargetState = IMP_STATE_SCARED;
 					break;
+				}
 
-				case IMP_STATE_ATTACK_1:
-				case IMP_STATE_ATTACK_2:
-					creature->MaxTurn = -1;
+				if (AI.bite && AI.distance < IMP_IDLE_ATTACK_RANGE && item->TriggerFlags < 10)
+				{
```

Comment:
At the end of the condition, is checking that the item->TriggerFlags < 10

TriggerFlags is the OCB, I the planned ocbs for this creature were 0,1, 2 and 3.  So this last condition case shall not happens, right?